pandas/core/indexes/base.py

```diff
@@ -886,7 +886,7 @@ def astype(self, dtype, copy=True):
 
         Parameters
         ----------
-        indices : list
+        indices : array-like
```

Comment:
i think Sequence[int] would be more helpful

Reply:
Happy to change, my only question would be if we want to be using type hints for user facing API. The numpy documentation for `take` just uses `array_like`

Reply:
> i think Sequence[int] would be more helpful

we shouldn't use typing notation for docstrings. https://pandas.pydata.org/pandas-docs/dev/development/contributing_docstring.html#parameter-types

and from that guide...

> If the exact type is not relevant, but must be compatible with a NumPy array, array-like can be specified. If Any type that can be iterated is accepted, iterable can be used:

but the guide doesn't mention sequence. 

maybe `sequence of int` is allowed, but I think matching numpy is fine.